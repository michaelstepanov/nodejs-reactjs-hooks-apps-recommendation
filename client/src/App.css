@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.form-group.required .control-label:after {
  content:"*";
  color:red;
}

.react-datepicker-container .react-datepicker-wrapper {
  display: block;
}

.has-error > .react-select > div:first-child,
.has-error input
{
  border-color: #dc3545;
}

.has-error label
{
  color: #dc3545!important;
}

.has-error .validation-message
{
  display: block;
  color: #dc3545!important;
}

.validation-message {
  display: none;
}

.app-icon {
  width: 180px;
}

ul li {
  cursor: pointer;
}

.card-horizontal {
  display: flex;
  flex: 1 1 auto;
}

.app-icon {
  padding-left: 1.25rem;
  padding-top: 1.25rem;
}
