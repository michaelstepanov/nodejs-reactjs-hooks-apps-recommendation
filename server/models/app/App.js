const Collection = require('../../classes/collection/Collection');

/**
 * Even though the class is empty it's used for filtering - filters may be defined for a specific Collection child class
 */
class App extends Collection {}

module.exports = App;
