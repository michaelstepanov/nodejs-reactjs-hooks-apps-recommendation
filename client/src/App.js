import React from 'react';
import './App.css';
import {GlobalProvider} from "./contexts/GlobalState";
import AppInfoContainer from "./components/AppInfo/AppInfoContainer";
import AppFilterFormContainer from "./components/AppFilterForm/AppFilterFormContainer";
import AppsContainer from "./components/Apps/AppsContainer";

function App() {
  return (
    <GlobalProvider>
      <div>
        <div className="container-fluid">
          <div className="page-header text-center mt-4 mb-4">
            <h1>Apps Recommendation</h1>
          </div>
          <div className="row">
            <div className="col-12 col-md-6 col-lg-4 mt-4">
              <h2>Filter</h2>

              <AppFilterFormContainer />
            </div>
            <div className="col-12 col-md-6 col-lg-3 mt-4">
              <h2>Apps</h2>

              <AppsContainer />
            </div>
            <div className="col-12 col-md-12 col-lg-5 mt-4">
              <h2>App Info</h2>

              <AppInfoContainer />
            </div>
          </div>
        </div>
      </div>
    </GlobalProvider>
  );
}

export default App;
