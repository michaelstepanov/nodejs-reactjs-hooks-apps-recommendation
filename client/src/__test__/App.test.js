import React from 'react';
import ReactDOM from 'react-dom';
import {cleanup} from '@testing-library/react';
import App from "../App";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
