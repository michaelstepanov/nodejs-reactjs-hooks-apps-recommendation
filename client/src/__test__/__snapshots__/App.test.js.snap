// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      className="page-header text-center mt-4 mb-4"
    >
      <h1>
        Apps Recommendation
      </h1>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-md-6 col-lg-4 mt-4"
      >
        <h2>
          Filter
        </h2>
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-text"
            >
              <form
                data-testid="form"
                onSubmit={[Function]}
              >
                <div
                  className="form-group required react-datepicker-container"
                >
                  <label
                    className="control-label"
                    htmlFor="birthdate"
                  >
                    Birthdate
                  </label>
                  <div
                    className="react-datepicker-wrapper"
                  >
                    <div
                      className="react-datepicker__input-container"
                    >
                      <input
                        autoComplete="off"
                        className="form-control"
                        disabled={false}
                        id="birthdate"
                        name="birthdate"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Select birthdate"
                        readOnly={false}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <small
                    className="form-text validation-message"
                  />
                </div>
                <div
                  className="form-group required"
                >
                  <label
                    className="control-label"
                    htmlFor="categories"
                  >
                    Categories
                  </label>
                  <div
                    className="react-select css-2b097c-container"
                    id="categories"
                    onKeyDown={[Function]}
                  >
                    <div
                      className=" css-yk16xz-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className=" css-1hwfws3"
                      >
                        <div
                          className=" css-1wa3eu0-placeholder"
                        >
                          Select max 3 categories
                        </div>
                        <div
                          className="css-b8ldur-Input"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-4-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "label": "input",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          className=" css-1okebmr-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          className=" css-tlfecz-indicatorContainer"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-6q0nyr-Svg"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div>
                      <input
                        name="categories"
                        type="hidden"
                      />
                    </div>
                  </div>
                  <small
                    className="form-text validation-message"
                  />
                </div>
                <div
                  className="form-group required"
                >
                  <label
                    className="control-label"
                    htmlFor="rating_gte"
                  >
                    Min Rating
                  </label>
                  <div
                    className="react-select css-2b097c-container"
                    id="rating_gte"
                    onKeyDown={[Function]}
                  >
                    <div
                      className=" css-yk16xz-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className=" css-1hwfws3"
                      >
                        <div
                          className=" css-1wa3eu0-placeholder"
                        >
                          Select min rating
                        </div>
                        <div
                          className="css-b8ldur-Input"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-5-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "label": "input",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          className=" css-1okebmr-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          className=" css-tlfecz-indicatorContainer"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-6q0nyr-Svg"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="rating_gte"
                      type="hidden"
                      value=""
                    />
                  </div>
                  <small
                    className="form-text validation-message"
                  />
                </div>
                <button
                  className="btn btn-primary pull-right"
                  data-testid="filter"
                  disabled={false}
                  type="submit"
                >
                  Filter
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-12 col-md-6 col-lg-3 mt-4"
      >
        <h2>
          Apps
        </h2>
        <div>
          No data
        </div>
      </div>
      <div
        className="col-12 col-md-12 col-lg-5 mt-4"
      >
        <h2>
          App Info
        </h2>
        <div>
          No data
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppFilterForm matches snapshot 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      className="page-header text-center mt-4 mb-4"
    >
      <h1>
        Apps Recommendation
      </h1>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-md-6 col-lg-4 mt-4"
      >
        <h2>
          Filter
        </h2>
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="card-text"
            >
              <form
                data-testid="form"
                onSubmit={[Function]}
              >
                <div
                  className="form-group required react-datepicker-container"
                >
                  <label
                    className="control-label"
                    htmlFor="birthdate"
                  >
                    Birthdate
                  </label>
                  <div
                    className="react-datepicker-wrapper"
                  >
                    <div
                      className="react-datepicker__input-container"
                    >
                      <input
                        autoComplete="off"
                        className="form-control"
                        disabled={false}
                        id="birthdate"
                        name="birthdate"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Select birthdate"
                        readOnly={false}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <small
                    className="form-text validation-message"
                  />
                </div>
                <div
                  className="form-group required"
                >
                  <label
                    className="control-label"
                    htmlFor="categories"
                  >
                    Categories
                  </label>
                  <div
                    className="react-select css-2b097c-container"
                    id="categories"
                    onKeyDown={[Function]}
                  >
                    <div
                      className=" css-yk16xz-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className=" css-1hwfws3"
                      >
                        <div
                          className=" css-1wa3eu0-placeholder"
                        >
                          Select max 3 categories
                        </div>
                        <div
                          className="css-b8ldur-Input"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-2-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "label": "input",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          className=" css-1okebmr-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          className=" css-tlfecz-indicatorContainer"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-6q0nyr-Svg"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div>
                      <input
                        name="categories"
                        type="hidden"
                      />
                    </div>
                  </div>
                  <small
                    className="form-text validation-message"
                  />
                </div>
                <div
                  className="form-group required"
                >
                  <label
                    className="control-label"
                    htmlFor="rating_gte"
                  >
                    Min Rating
                  </label>
                  <div
                    className="react-select css-2b097c-container"
                    id="rating_gte"
                    onKeyDown={[Function]}
                  >
                    <div
                      className=" css-yk16xz-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className=" css-1hwfws3"
                      >
                        <div
                          className=" css-1wa3eu0-placeholder"
                        >
                          Select min rating
                        </div>
                        <div
                          className="css-b8ldur-Input"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-3-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "label": "input",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          className=" css-1okebmr-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          className=" css-tlfecz-indicatorContainer"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-6q0nyr-Svg"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="rating_gte"
                      type="hidden"
                      value=""
                    />
                  </div>
                  <small
                    className="form-text validation-message"
                  />
                </div>
                <button
                  className="btn btn-primary pull-right"
                  data-testid="filter"
                  disabled={false}
                  type="submit"
                >
                  Filter
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-12 col-md-6 col-lg-3 mt-4"
      >
        <h2>
          Apps
        </h2>
        <div>
          No data
        </div>
      </div>
      <div
        className="col-12 col-md-12 col-lg-5 mt-4"
      >
        <h2>
          App Info
        </h2>
        <div>
          No data
        </div>
      </div>
    </div>
  </div>
</div>
`;
